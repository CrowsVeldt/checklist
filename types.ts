export type ChecklistType = {
  id: string;
  title: string;
  entries: ChecklistEntryType[] | [];
};

export type ChecklistEntryType = {
  id: string;
  status: boolean;
  title: string;
  required: boolean;
  childOf: ChecklistEntryType | undefined;
  parentTo: ChecklistEntryType | ChecklistEntryType[] | undefined;
};
