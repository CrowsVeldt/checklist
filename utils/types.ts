export type ChecklistType = {
  id: string;
  title: string;
  entries: ChecklistEntryType[] | [];
};

export type ChecklistEntryType = {
  id: string;
  status: boolean;
  title: string;
  required: boolean;
  parentTo: ChecklistEntryType[];
};
