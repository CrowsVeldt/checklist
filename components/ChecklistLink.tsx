import { ChecklistType } from "@/utils/types";
import { router } from "expo-router";
import { Pressable, StyleSheet, Text, View } from "react-native";
import { AntDesign } from "@expo/vector-icons";

export default function ChecklistLink({
  list,
  triggerModal,
}: {
  list: ChecklistType;
  triggerModal: any;
}) {
  return (
    <Pressable
      onPress={() =>
        router.push({
          pathname: "/checklist/[checklist]",
          params: { checklist: list.id },
        })
      }
      style={({ pressed }) =>
        pressed ? [styles.link, styles.linkPressed] : styles.link
      }
    >
      <Text style={styles.listTitle}>{list.title}</Text>
      <View style={styles.buttonContainer}>
        <Pressable
          style={({ pressed }) =>
            pressed ? [styles.button, styles.linkPressed] : styles.button
          }
          onPress={() =>
            router.push({
              pathname: "/edit",
              params: { id: list.id },
            })
          }
        >
          <AntDesign name="edit" size={24} color="black" />
        </Pressable>
        <Pressable
          style={({ pressed }) =>
            pressed ? [styles.button, styles.linkPressed] : styles.button
          }
          onPress={() => triggerModal(list)}
        >
          <AntDesign name="delete" size={24} color="black" />
        </Pressable>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  link: {
    flexDirection: "row",
    borderColor: "black",
    borderWidth: 1,
    width: "100%",
    height: 50,
    justifyContent: "space-between",
    alignItems: "center",
  },
  linkPressed: {
    backgroundColor: "gray",
  },
  listTitle: {
    marginStart: 30,
    fontSize: 25,
  },
  button: {
    height: 50,
    width: 50,
    justifyContent: "center",
    alignItems: "center",
    borderColor: "black",
    borderWidth: 1,
  },
  buttonContainer: {
    flexDirection: "row",
  },
});
