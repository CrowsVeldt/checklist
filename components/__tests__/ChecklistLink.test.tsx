import { render } from "@testing-library/react-native";
import ChecklistLink from "../ChecklistLink";
import { ChecklistType } from "@/utils/types";

const testList: ChecklistType = {
  id: "123abc",
  title: "default checklist",
  entries: [
    {
      id: "456def",
      status: false,
      title: "item 1",
      required: true,
      childOf: undefined,
      parentTo: undefined,
    },
    {
      id: "789ghi",
      status: false,
      title: "item 2",
      required: true,
      childOf: undefined,
      parentTo: undefined,
    },
  ],
};

describe("<ChecklistLink />", () => {
  test("ChecklistLink renders correctly", () => {
    const tree = render(<ChecklistLink list={testList} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
