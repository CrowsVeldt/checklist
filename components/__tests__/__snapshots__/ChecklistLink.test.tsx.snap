// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChecklistLink /> ChecklistLink renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "black",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "marginStart": 30,
      }
    }
  >
    default checklist
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "marginStart": "auto",
        "width": 40,
      }
    }
  >
    <Text />
  </View>
</View>
`;
